agent/engine/errors.go

```diff
@@ -137,3 +125,133 @@ func (err TaskStoppedBeforePullBeginError) Error() string {
 func (TaskStoppedBeforePullBeginError) ErrorName() string {
 	return "TaskStoppedBeforePullBeginError"
 }
+
+// CannotStopContainerError indicates any error when trying to stop a container
+type CannotStopContainerError struct {
```

Comment:
I like that idea. I'll keep working on it, don't want to delay merging this PR on that. Let me know if everything else looks fine.